codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerEnumGenerator.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.StringShape
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.server.smithy.ServerRuntimeType
+import software.amazon.smithy.rust.codegen.smithy.CodegenMode
+import software.amazon.smithy.rust.codegen.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.generators.EnumGenerator
+import software.amazon.smithy.rust.codegen.util.dq
+
+class ServerEnumGenerator(
+    model: Model,
+    symbolProvider: RustSymbolProvider,
+    private val writer: RustWriter,
+    shape: StringShape,
+    enumTrait: EnumTrait,
+    mode: CodegenMode,
+    private val runtimeConfig: RuntimeConfig,
+) : EnumGenerator(model, symbolProvider, writer, shape, enumTrait, mode,) {
+    override fun renderFromForStr() {
+        val errorStruct = "${enumName}UnknownVariantError"
+        writer.write("##[derive(Debug, PartialEq, Eq, Hash)]")
+        writer.write("pub struct $errorStruct(String);")
+        writer.rustBlock("impl #T<&str> for $enumName", RuntimeType.TryFrom) {
+            write("type Error = $errorStruct;")
+            writer.rustBlock("fn try_from(s: &str) -> Result<Self, <$enumName as #T<&str>>::Error>", RuntimeType.TryFrom) {
+                writer.rustBlock("match s") {
+                    sortedMembers.forEach { member ->
+                        write("""${member.value.dq()} => Ok($enumName::${member.derivedName()}),""")
```

Comment:
`dq` is useful when your enclosing string is in double quotes:

```suggestion
                        write("${member.value.dq()} => Ok($enumName::${member.derivedName()}),")
```

When using `"""`, you can write double quotes directly:

```suggestion
                        write(""""${member.value}" => Ok($enumName::${member.derivedName()}),""")
```

But using `"""` _and_ `dq` is noisy. I notice this has been copied from `EnumGenerator`, we should make the improvement there too.